### ISSUE TITLE
Replace Open Babel usage with RDKit

### ISSUE THREAD

**Issue by lilleswing** [deepchem/deepchem#371]

**Comment by rbharath** [comment]
+1. Openbabel has caused a number of segfaults and issues, so would be good to see if we can get RDKit to fill this role. PRs are welcome. The key places to replace would be `dc.feat.GridFeaturizer` and `dc.feat.hydrogenate_and_compute_partial_charges`.

**Comment by proteneer** [comment]
Any idea on the partial charge model used in OpenBabel? RDKit afaik only supports the Gasteiger, which are pretty terrible FWIW (though @greglandrum) may know more.

Also, the 3D conformer generator in RDKit has been improved significantly, namely with Sereina's ETKDG model, which can be invoked very easily.

**Comment by rbharath** [comment]
Our handling of charges/protonation is uniformly terribly right now (chime in on #368 with ideas!), so we're pretty open to establishing new best practices :-)

**Comment by proteneer** [comment]
Based on the docstring of hydrogenate_and_compute_partial_charges:

```
Outputs a hydrogenated pdb and a pdbqt with partial charges.
  Takes an input file in specified format. Generates two outputs:
  -) A pdb file that contains a hydrogenated (at pH 7.4) version of
     original compound.
  -) A pdbqt file that has computed Gasteiger partial charges. This pdbqt
     file is build from the hydrogenated pdb.
```

So it does look like it's doing Gasteiger for partial charges. I have no idea what's its doing for the protonation states. Presumably it's just looking up the pKa tables for the various amino acids, so it probably shouldn't be too difficult to replace.

**Comment by greglandrum** [comment]
As @proteneer already pointed out: the RDKit has Gasteiger charges, which are good for what they are good for.
There are also MMFF94 charges:
```
In [9]: m = Chem.AddHs(Chem.MolFromSmiles('CCOC'))

In [10]: fps = AllChem.MMFFGetMoleculeProperties(m)

In [11]: [fps.GetMMFFPartialCharge(x) for x in range(m.GetNumAtoms())]
Out[11]: [0.0, 0.28, -0.56, 0.28, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
```
which, I am sure, have their own set of pathologies. Plus they will only be available for atoms that have MMFF94 parameters.

**Comment by evanfeinberg** [comment]
Confirming that in our initial development of this we used Gasteiger for
partial charges. In the wise words of Knuth ```Premature optimization is
the root of all evil``` -- we figured we'd just get something to work and
optimize later. Now that it's "later"...I'm impressed with OPLS's
improvements and results as of late. Can we integrate OPLS's charge models?

PS +1 to rdkit...far easier to package and install. without newer GCC
```openbabel``` can be a major installation headache.
